### PR TITLE
add more functions to aiosource

### PR DESCRIPTION
**Pull request by KPostOffice**
Some of the functions that were missing from `aiosource` can't be called by the super class `source` because their dependent functions are already implemented in an asynchronous function which needs to be awaited to get results. Without calling `await` we only get a `coroutine` object.

### CODE REVIEW

File: thoth/python/aiosource.py [hunk 1]
```diff
@@ -268,3 +268,44 @@ async def get_package_versions(self, package_name: str) -> AsyncIterableVersions
...
+    async def get_package_artifacts(self, package_name: str, package_version: str):
+        """Return list of artifacts corresponding to package name and package version."""
+        to_return = []
+        possible_continuations = [".win32", ".tar.gz", ".whl", ".zip", ".exe", ".egg", "-"]
+        async for artifact_name, artifact_url in await self._simple_repository_list_artifacts(package_name):
+            # Convert all artifact names to lowercase - as a shortcut we simply convert everything to lowercase.
+            artifact_name.lower()
+
+            for i in possible_continuations:
+                if artifact_name.startswith(f"{package_name}-{package_version}{i}"):
```
Comment (fridex) [inline thoth/python/aiosource.py:281]: Does this work also with platform tags in wheel distributions?

https://www.python.org/dev/peps/pep-0425/

Comment (KPostOffice) [reply thoth/python/aiosource.py:281]: Yes because `-` is included in possible continuations

Comment (fridex) [reply thoth/python/aiosource.py:281]: Will it match for example `tensorflow-2.2.0-cp36-cp36m-manylinux2010_x86_64.whl`?

https://pypi.org/project/tensorflow/#files

Comment (KPostOffice) [reply thoth/python/aiosource.py:281]: Yeah, that starts with `tensorflow-2.2.0-`

Comment (fridex) [reply thoth/python/aiosource.py:281]: Ah, I didn't see the `-` trick :)

**Pull request merged**